subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/DaemonBuildActionExecuter.java

```diff
@@ -40,7 +42,14 @@ public BuildActionResult execute(BuildAction action, ConnectionOperationParamete
         ClassPath classPath = DefaultClassPath.of(operationParameters.getInjectedPluginClasspath());
 
         DaemonParameters daemonParameters = parameters.getDaemonParameters();
-        BuildActionParameters actionParameters = new DefaultBuildActionParameters(daemonParameters.getEffectiveSystemProperties(), daemonParameters.getEnvironmentVariables(), SystemProperties.getInstance().getCurrentDir(), operationParameters.getBuildLogLevel(), daemonParameters.isEnabled(), classPath);
+        Map<String, String> effectiveSystemProperties;
+        Map<String, String> operationParametersSystemProperties = operationParameters.getSystemProperties(null);
+        if (operationParametersSystemProperties != null) {
+            effectiveSystemProperties = daemonParameters.getEffectiveSystemProperties(operationParametersSystemProperties);
+        } else {
+            effectiveSystemProperties = daemonParameters.getEffectiveSystemProperties();
+        }
```

Comment:
Can we not do something like this?
```suggestion
        Map<String, String> operationParametersSystemProperties = operationParameters.getSystemProperties(System.getProperties());
        Map<String, String> effectiveSystemProperties = daemonParameters.getEffectiveSystemProperties(operationParametersSystemProperties);
```

Reply:
I don't think that's correct. System.getProperties() returns an object of type `Properties` which is not a `Map<String, String>`.